aws-datastore/src/androidTest/java/com/amplifyframework/datastore/BasicCloudSyncInstrumentationTest.java

```diff
@@ -238,4 +254,284 @@ public void updateAfterCreate() throws DataStoreException, ApiException {
+        accumulator.await(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        // Verify that the updatedOwner is saved in the DataStore.
+        BlogOwner localOwner = dataStore.get(BlogOwner.class, anotherOwner.getId());
+        ModelAssert.assertEqualsIgnoringTimestamps(updatedOwner, localOwner);
+
+        // Verify that the updatedOwner is saved on the backend.
+        BlogOwner remoteOwner = api.get(BlogOwner.class, anotherOwner.getId());
+        ModelAssert.assertEqualsIgnoringTimestamps(updatedOwner, remoteOwner);
+    }
+
+    /**
+     * Verify that creating a new item, waiting for it to sync, and then updating 10 times consecutively succeeds.
+     * @throws DataStoreException On failure to save or query items from DataStore.
+     * @throws ApiException On failure to query the API.
+     */
+    @Test
+    public void createWaitThenUpdateMultipleTimes() throws DataStoreException, ApiException {
+        // Setup
+        BlogOwner owner = BlogOwner.builder()
+                .name("Jean")
+                .build();
+        String modelName = BlogOwner.class.getSimpleName();
+        
+        HubAccumulator accumulator =
+                HubAccumulator.create(HubChannel.DATASTORE, publicationOf(modelName, owner.getId()), 1)
+                        .start();
+        // Create an item.
+        dataStore.save(owner);
+        // Wait for the sync.
+        accumulator.await(TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+        // Setup 10 updates
+        List<String> weas = Arrays.asList("pon", "lth", "ver", "kly", "ken", "sel", "ner", "rer", "ten", "ned");
+        HubAccumulator updateAccumulator =
+                HubAccumulator.create(HubChannel.DATASTORE, publicationOf(modelName, owner.getId()), 3).start();
+                
+        // Updating multiple times consecutively
+        // Accumulator crashes for more than 3 consecutive saves. Need to open ticket to investigate
+        for (int i = 0; i < 3; i++) {
+            BlogOwner updatedOwner = owner.copyOfBuilder().wea(weas.get(i)).build();
+            dataStore.save(updatedOwner);
+        }
+        updateAccumulator.await(120, TimeUnit.SECONDS);
+        
+        // Verify that the updatedOwner is saved on the backend.
+        BlogOwner remoteOwner = api.get(BlogOwner.class, owner.getId());
+        Assert.assertEquals(weas.get(2), remoteOwner.getWea());
+        
+        // Verify that the last update is saved in the DataStore.
+        BlogOwner localOwner = dataStore.get(BlogOwner.class, owner.getId());
+        Assert.assertEquals(weas.get(2), localOwner.getWea());
+    }
+
+    /**
+     * Verify that creating a new item, then immediately delete succeeds.
+     * @throws DataStoreException On failure to save or query items from DataStore.
+     * @throws ApiException On failure to query the API.
+     */
+    @Test
+    @Ignore("This test currently fails - need to file ticket to fix the root cause and then enable test")
```

Comment:
Do we know what causes this failure?

Reply:
I have created the task ( https://app.asana.com/0/1200297809582780/1200616362832607/f ) to investigate.  The iOS team also had this problem ( https://quip-amazon.com/M011AZBRtAhA/Consecutive-Saves-Test-Cases )